src/LieTheory/WeylGroup.jl

```diff
@@ -75,11 +75,56 @@ Construct a Weyl group element from the given word.
 The word must be a list of integers, where each integer is the index of a simple reflection.
 
 If the word is known to be in short lex normal form, the normalization can be skipped by setting `normalize=false`.
+
+# Examples
+```jldoctest
+julia> W = weyl_group(:A, 2);  wrd = [1, 2, 1];
+
+julia> x = W(wrd)
+s1 * s2 * s1
+
+julia> word(x) == wrd
+true
+```
 """
 function (W::WeylGroup)(word::Vector{<:Integer}; normalize::Bool=true)
   return WeylGroupElem(W, word; normalize)
 end
 
+@doc raw"""
+    (W::WeylGroup)(sylls::AbstractVector{Pair{T, S}}; normalize::Bool=true) where {T <: Integer, S <: IntegerUnion} -> WeylGroupElem
+
+Construct a Weyl group element from the given syllables.
+
+The syllables must be a list of pairs, where each entry is the index of a simple reflection and its exponent.
+
+If the syllables describe a word in short lex normal form, the normalization can be skipped by setting `normalize=false`.
+
+# Examples
+```jldoctest
+julia> W = weyl_group(:A, 2);  pairs = [1 => 1, 2 => 1];
+
+julia> x = W(pairs)
+s1 * s2
+
+julia> syllables(x) == pairs
+true
+```
+"""
+function (W::WeylGroup)(sylls::AbstractVector{Pair{T, S}}; normalize::Bool=true) where {T <: Integer, S <: IntegerUnion}
+  n = ngens(W)
+  res = Vector{T}()
+  
+  for pair in sylls
+    @req 0 < pair.first && pair.first <= n "generator number is at most $n"
+    if pair.second != 0
+      push!(res, pair.first)
+    end
+  end
+
```

Comment:
```suggestion
  res = T[pair.first for pair in sylls if isodd(pair.second)]
```
You don't need to check if all generator indices are valid, since the constructor `WeylGroupElem` already does that.

It is wrong to just check for `pair.second != 0`, as all generators are self-inverse, and thus an exponent of 2  (or rather any even number) can be dropped.

Reply:
Ah that makes sense, thank you. Great usage for `isodd`!